src/commands/plugins.rs

```diff
@@ -334,10 +335,30 @@ pub struct List {
 
 impl List {
     pub async fn run(self) -> Result<()> {
-        let mut plugins = if self.installed {
-            Self::list_installed_plugins()
-        } else {
-            Self::list_catalogue_and_installed_plugins().await
+        ListCommandCore {
+            which: WhichPlugins::Installed,
+            filter: self.filter,
+        }
+        .run()
+        .await
+    }
+}
+
+enum WhichPlugins {
+    Installed,
+    All,
+}
+
+struct ListCommandCore {
+    which: WhichPlugins,
+    filter: Option<String>,
+}
+
+impl ListCommandCore {
+    pub async fn run(self) -> Result<()> {
+        let mut plugins = match &self.which {
+            WhichPlugins::Installed => Self::list_installed_plugins(),
```

Comment:
Now that `list` only shows installed, I wonder if it is redundant to keep the `[installed]` labels next to each listing?
```
$ spin plugins list
check-for-update 0.1.0 [installed]
cloud 0.4.0 [installed]
js2wasm 0.6.0post.1694728162 [installed]
platform 0.1.0post.1694720740 [installed]
pluginify 0.6.0 [installed]
py2wasm 0.3.0post.1693957212 [installed]
trigger-sqs 0.3 [installed]
```

Reply:
We could.  Keeping it keeps the presentations unified, but we can remove it if people feel strongly.